gio/src/dbus_connection.rs

```diff
@@ -515,4 +725,89 @@ impl DBusConnection {
             );
         }
     }
+
+    // rustdoc-stripper-ignore-next
+    /// Subscribe to a D-Bus signal and receive signal emissions as a stream.
+    ///
+    /// See [`Self::signal_subscribe`] for arguments.  `map_signal` maps the
+    /// received signal to the stream's element.  If it returns `None` the
+    /// signal emission is not propagated downstream.
+    ///
+    /// The returned stream holds a strong reference to this D-Bus connection,
+    /// and unsubscribes from the signal when dropped.
+    ///
+    /// To avoid reference cycles you may wish to downgrade the returned
+    /// stream to hold only weak reference to the connection using
+    /// [`SubscribedSignalStream::downgrade`].
+    pub fn receive_signal<T: 'static, F: Fn(DBusSignalRef) -> Option<T> + 'static>(
+        &self,
+        sender: Option<&str>,
+        interface_name: Option<&str>,
+        member: Option<&str>,
+        object_path: Option<&str>,
+        arg0: Option<&str>,
+        flags: DBusSignalFlags,
+        map_signal: F,
+    ) -> SubscribedSignalStream<SignalSubscription, impl Stream<Item = T> + use<T, F>> {
+        let (tx, rx) = mpsc::unbounded();
```

Comment:
@sdroege I'm not sure how to do back-pressure here if signals arrive faster than the app consumes them from the channel; currently I just keep filling up the channel until the processes runs out of memory.  That's easy to implement, but probably not a good idea?

Reply:
The only other options you have are dropping once a threshold is reached, or blocking.

Do you know how this is handled in zbus? Probably "blocking" (asynchronously)?

Reply:
Yeah blocking the main context is not going to be very useful indeed, especially as the receiving end of the channel is likely handled on the main context too :)

I don't really have any suggestions here. Maybe just keep it like this and document it carefully, and then we'll see how it works in practice?